addons/website/static/src/snippets/s_chart/options.js

```diff
@@ -1,85 +1,132 @@
 /** @odoo-module **/
 
 import { _t } from "@web/core/l10n/translation";
-import options from "@web_editor/js/editor/snippets.options.legacy";
-import weUtils from "@web_editor/js/common/utils";
 import { isCSSColor } from '@web/core/utils/colors';
+import weUtils from "@web_editor/js/common/utils";
+import {
+    SnippetOption,
+    SnippetOptionComponent,
+} from "@web_editor/js/editor/snippets.options";
+import {
+    registerWebsiteOption,
+} from "@website/js/editor/snippets.options";
+
+class InnerChartOptionComponent extends SnippetOptionComponent {
+    static themeArray = ['o-color-1', 'o-color-2', 'o-color-3', 'o-color-4', 'o-color-5'];
+    onAddDataset() {
+        const usedColor = this.renderContext.datasets.map(dataset => dataset.backgroundColor);
+        const color = this.constructor.themeArray.filter(el => !usedColor.includes(el))[0] || this._randomColor();
+        this.renderContext.datasets.push({
+            label: "",
+            data: [],
+            backgroundColor: color,
+            borderColor: color,
+        });
+        this.env.snippetOption._reloadGraph();
+    }
+    onAddRow() {
+        this.renderContext.labels.push("");
+        this.env.snippetOption._reloadGraph();
+    }
+    onRemoveDataset(datasetIndex) {
+        this.renderContext.datasets.splice(datasetIndex, 1);
+        this.env.snippetOption._reloadGraph();
+        this._refreshDatasetPalette()
+    }
+    onRemoveRow(labelIndex) {
+        this.renderContext.labels.splice(labelIndex, 1);
+        for (const dataset of this.renderContext.datasets) {
+            dataset.data.splice(labelIndex, 1);
+        }
+        this.env.snippetOption._reloadGraph();
+    }
+    onCellFocus(datasetIndex, labelIndex) {
+        this.renderContext.focusedDatasetIndex = datasetIndex;
+        this.renderContext.focusedLabelIndex = labelIndex;
+        this._refreshDatasetPalette()
+    }
+    setDatasetLabel(datasetIndex, datasetLabel) {
+        this.renderContext.datasets[datasetIndex].label = datasetLabel;
+        this.env.snippetOption._reloadGraph();
```

Comment:
Indeed... and it was not before either...
except "add row" and "add column" which did get an history step recorded - but did not refresh the DOM upon undo ("removes" did not have history steps either - since the public widget is not re-started upon undo)